tests/test_ts3_monitor.py

```diff
@@ -0,0 +1,66 @@
+"""Test suite for monitoring_scripts.ts3_monitor"""
+
+import pytest
+
+from monitoring_scripts import ts3_monitor as unit
+from monitoring_scripts.nagios_common import Codes
+
+
+def run_and_assert(capfd=None, expected_code=Codes.OK, expected_message_part=None,
+                   host='ts.some.server',
+                   port=9987, timeout=10, debug=False):
+    """Run ts3_monitor.main with the specified parameters and asserts that
+    its output and return code match the expected return code and Nagios
+    guidelines"""
+    
+    with pytest.raises(SystemExit) as exit_info:
+        unit.main(host=host, port=port, timeout=timeout, debug=debug)
+    
+    assert exit_info.value.code == expected_code.value
+
+    if(capfd):
+        out = capfd.readouterr()[0]
+        assert out.startswith(expected_code.name + ':')
+
+        if(expected_message_part):
+            assert expected_message_part in out
+    
+    
+@pytest.mark.parametrize(
+    'ts3server',
+    [
+        'invalidhostname',
+        '260.1.1.1'
+    ]
+)
+def test_invalid_host(capfd, ts3server):
```

Comment:
This relies on the given hostname to be actually unresolvable on the running machine. The safer approach would be to mock the socket so that it raises `gaierror`.